### PR TITLE
Updates to ARRM10to60E2r1 mesh

### PR DESCRIPTION
**Pull request by darincomeau**
This brings in additional commits for this mesh support that were added as a result of review of https://github.com/E3SM-Project/E3SM/pull/5178.

- Reverts 15 min sea ice timestep for G-cases (now uses default 30 min)
- Changes `Redi_min_layers_diag` to 15 for this mesh only
- Updates name of `analysis_mask_file`
- Adds land `finidat` file for `WCYCL1950` compset with `ne30pg2` atm

### CODE REVIEW

**Review by apcraig — APPROVED** [review]

**Comment by darincomeau** [comment]
This will be closed as redundant once https://github.com/darincomeau/E3SM/pull/28 is merged.

A follow on PR will be made to update `finidat` and `fsurdat` for `arcticx4v1` 1950 conditions from here (testing now): https://github.com/E3SM-Project/E3SM/tree/darincomeau/lnd/add-arcticx4v1-land-1950-files

**Comment by darincomeau** [comment]
I just brought in the commits to set the land files for `WCYCL1950.arcticx4v1_ARRM60to10E2r1` configuration.

@milenaveneziani I think this is ready to merge to keep this branch up to date?

**Comment by darincomeau** [comment]
hmm nevermind - conflicts showing up now. I didn't rebase before adding the new commits

**Comment by darincomeau** [comment]
Ok now this looks correct

File: driver-mct/cime_config/config_component_e3sm.xml [hunk 1]
```diff
@@ -372,7 +372,6 @@
       <value compset="_DATM.*_SLND.*MPAS" grid="oi%oRRS15to5">96</value>
       <value compset="_DATM.*_SLND.*MPAS" grid="oi%oARRM60to10">96</value>
       <value compset="_DATM.*_SLND.*MPAS" grid="oi%oARRM60to6">96</value>
-      <value compset="_DATM.*_SLND.*MPAS" grid="oi%ARRM10to60">96</value>
```
Comment (milenaveneziani) [inline driver-mct/cime_config/config_component_e3sm.xml:375]: you are removing this one for now because we want to Hailong first, right?

**Review by milenaveneziani — APPROVED** [review]
Looks good @darincomeau. Only a small confusion about ATM_NCPL, but that's probably just my problem.

File: driver-mct/cime_config/config_component_e3sm.xml (hunk 1 above)
Comment (darincomeau) [reply driver-mct/cime_config/config_component_e3sm.xml:375]: this came in [1199ea1](https://github.com/darincomeau/E3SM/pull/26/commits/1199ea104fb74e37f8b937e589909ce2087a0e77) to change the default timestep back to 30 minutes for G-cases from our discussions with Robert a few weeks ago.

But these leaves in place `ATM_NCPL=96` for the `arcticx4v1` atm mesh on this branch (but we haven't opened the companion PR into E3SM master yet - and needs discussion as noted).

**Comment by darincomeau** [comment]
I built (but did not run) some cases to make sure nothing unexpected changed in `GMPAS-JRA1p4.TL319_ARRM10to60E2r1`, `WCYCL1950.ne30pg2_ARRM10to60E2r1`, and `WCYCL1950.arcticx4v1pg2_ARRM10to60E2r1`; confirmed right land IC files, timesteps/coupling frequencies (though `arcticx4v1pg2_ARRM10to60E2r1` still needs namelist change for ocn timestep to 7:30).

**Pull request merged**